src/document/document.ts

```diff
@@ -618,7 +618,10 @@ export class Document<T> {
     }
 
     if (changes.length && this.eventStreamObserver) {
-      this.eventStreamObserver.next({
+      // RemoteChange event should be sent synchronously.
+      // Because the next change should be applied after this event.
+      // application of the next change is blocked until this event is processed.
```

Comment:
```suggestion
      // NOTE: RemoteChange event should be emitted synchronously with
      // applying changes. This is because 3rd party model should be synced
      // with the Document after RemoteChange event is emitted. If the event
      // is emitted asynchronously, the model can be changed and breaking 
      // consistency.
```